pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,347 @@
+
+    return objmgr_inst
+
+
+def set_server_statistics(context, desired_state):
+    """
+    Set the server statistics state to on or off, enabling or disabling
+    the statistics.
+
+    Parameters:
+      desired_state (:class": `boolean)
+        Value desired for the GatherStasticalData property
+    """
+    conn = context.conn
+    interop_ns = context.wbem_server.interop_ns
+
+    objmgr = get_objmgr_inst(context)
+    cur_state = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+    if cur_state == desired_state:
+        click.echo("Server already in state {}".format(desired_state))
+        return
+
+    objmgr[OBJMGR_STAT_PROPERTY_NAME] = desired_state
+
+    # Otherwise modify the property in the server
+    try:
+        conn.ModifyInstance(objmgr, PropertyList=[OBJMGR_STAT_PROPERTY_NAME],
+                            IncludeQualifiers=False)
+    except Error as ce:
+        raise click.ClickException('Modification of GatherStasticalData '
+                                   'property on server failed {}'.format(ce))
+
+    modified_objmgr = conn.GetInstance(objmgr.path)
+
+    if modified_objmgr[OBJMGR_STAT_PROPERTY_NAME] != desired_state:
+        raise click.ClickException(
+            'Modification failed classname {} on server namespace {}'.
+            format(OBJMGR_CLN, interop_ns))
+
+    click.echo("Server GatherStatisticalData set {}". format(desired_state))
+
+
+def test_stats_enabled(context):
+    """
+    Execute ClickException of timestats not enabled
+    """
+    if not context.timestats:
+        raise click.ClickException(
+            'Pywbemcli time statistics not enabled. Use general options '
+            '-T/--timestats to enable statistics.')
+
+###############################################################
+#         Statistics cmds action functions
+###############################################################
+
+
+def cmd_statistics_server_on(context):
+    """
+    Attempt to enable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, True)
+
+
+def cmd_statistics_server_off(context):
+    """
+    Attempt to disable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, False)
+
+
+def cmd_statistics_status(context):
+    """
+    Show current status of both local and server statistics
+    """
+    test_conn_exists(context)
+
+    objmgr = get_objmgr_inst(context)
+    status = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+
+    local_stats = context.timestats
+    click.echo("Statistics state client={}; Server={}".format(local_stats,
+                                                              status))
+
+
+def cmd_statistics_reset(context):
+    """
+    Resets the client statistics. This command does not impact the the
+    statistics coming from the server.
```

Comment:
It does impact the server response time coming from the server, but not the server statistics.

Reply:
Clarified.